src/System.CommandLine/Help/HelpBuilder.cs

```diff
@@ -17,11 +17,13 @@ public class HelpBuilder : IHelpBuilder
         private Dictionary<ISymbol, Customization> Customizations { get; } = new();
 
         protected Resources Resources { get; }
+        protected ParseResult? ParseResult { get; }
         public int MaxWidth { get; }
 
-        public HelpBuilder(Resources resources, int maxWidth = int.MaxValue)
+        public HelpBuilder(Resources resources, int maxWidth = int.MaxValue, ParseResult? parseResult = null)
```

Comment:
It will make more sense to add the `ParseResult` parameter to the `IHelpBuilder.Write` method. Otherwise, this won't work with custom help builders.